### PR TITLE
Added DEPTH support for Amber Script.

### PR DESCRIPTION
**Pull request by asuonpaa**

### CODE REVIEW

File: src/vulkan/graphics_pipeline.cc [hunk 1]
```diff
@@ -113,9 +113,10 @@ VkCompareOp ToVkCompareOp(CompareOp op) {
       return VK_COMPARE_OP_GREATER_OR_EQUAL;
     case CompareOp::kAlways:
       return VK_COMPARE_OP_ALWAYS;
+    default:
```
Comment (dj2) [inline src/vulkan/graphics_pipeline.cc:116]: We've seen issues on compilers where there is no return even if the switch is fully covered. Please move this back out of the switch.

Comment (asuonpaa) [reply src/vulkan/graphics_pipeline.cc:116]: But then I'm getting an error of not covering the new kUnknown value in a switch. Should I do both then?

Comment (dj2) [reply src/vulkan/graphics_pipeline.cc:116]: Yea, add
```
case CompareOp::kUnknown:
  break;
```

and let the external return handle it.

**Comment by alan-baker** [comment]
FYI, The linux bot failures will need #844 to work.

**Pull request merged**